sklearn/metrics/pairwise.py

```diff
@@ -1420,8 +1419,13 @@ def pairwise_distances(X, Y=None, metric="euclidean", n_jobs=None, **kwds):
             raise TypeError("scipy distance metrics do not"
                             " support sparse matrices.")
 
-        dtype = bool if metric in PAIRWISE_BOOLEAN_FUNCTIONS else None
-        X, Y = check_pairwise_arrays(X, Y, dtype=dtype)
```

Comment:
Why are you introducing these changes? This isn't related to removing `warn_on_dtype`

Reply:
yes I agree but for the implementation in master for any metric in  `PAIRWISE_BOOLEAN_FUNCTIONS` the `check_pairwise_arrays` method type casts X and Y to boolean (which is what is passed as dtype)